### PR TITLE
[DE-424] Feat/log artist metrics with mlflow

### PR DESCRIPTION
**Pull request by lmontier-pass**
# DS PR

## Describe your changes

Log Metrics in MLflow

## Jira ticket number and/or notion link

https://www.notion.so/passcultureapp/880351701699470cb129e4e0ad6bbcba?v=cfbe4b16778b47218f252bd1b5add4a0&p=596ca17916e646e095dd724d449f71c2&pm=s

### Type of change
- [ ] hotfix (non-breaking change which fixes an issue)
- [ ] New model
- [ ] Training
- [ ] New features
- [ ] Bug Fix
- [ ] Code Refacto
- [ ] Performance Improvements
- [ ] Test
- [ ] CI
- [ ] Config
- [ ] Metrics

      
### Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] My code passes CI/CD tests
- [ ] I updated README.md
- [ ] I have updated the dag
- [ ] If my changes concern incremental table, I have altered their schema to accomodate with field's creation/deletion
- [ ] I have documented the corresponding [notion page](https://www.notion.so/passcultureapp/Team-Data-engineering-Data-science-22ab0eb5ddf34dc2a854d9f0e596e91b)
- [ ] I have made corresponding changes to the [tables documentation](https://www.notion.so/passcultureapp/Documentation-Tables-175a397a8e854ff4a55ae4f3620dbe3b)
- [ ] I have made corresponding changes to the [fields glossary](https://www.notion.so/passcultureapp/854a436a8f1541e1b6ec2a65f8bab600?v=798024ba90404b139e5a17407a3bc604)
- [ ] I will create a review on slack and ensure to specify the duration of the review task: short (<10min), medium (<30min), long (>30min)

### Added tests?
- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [x] ⏰ no, but I created a ticket

### CODE REVIEW

File: jobs/ml_jobs/artist_linkage/utils/constants.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import os
+
+GCP_PROJECT_ID = os.environ.get("GCP_PROJECT_ID", "passculture-data-ehp")
```
Comment (cdelabre) [inline jobs/ml_jobs/artist_linkage/utils/constants.py:3]: use common configs

File: jobs/ml_jobs/artist_linkage/utils/constants.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+ENV_SHORT_NAME = os.environ.get("ENV_SHORT_NAME", "dev")
+
+MLFLOW_EHP_URI = "https://mlflow.staging.passculture.team/"
+MLFLOW_PROD_URI = "https://mlflow.passculture.team/"
+MLFLOW_SECRET_NAME = "mlflow_client_id"
+
+EXPERIMENT_BASE_NAME = "artist_linkage"
```
Comment (cdelabre) [inline jobs/ml_jobs/artist_linkage/utils/constants.py:10]: Le mettre en paramètre du job avec une version plutôt

File: orchestration/dags/jobs/ml/link_artists.py [hunk 3]
```diff
@@ -29,6 +23,13 @@
...
 
+from airflow import DAG
+from airflow.models import Param
+from airflow.operators.python import PythonOperator
+from airflow.providers.google.cloud.transfers.gcs_to_bigquery import (
+    GCSToBigQueryOperator,
+)
+
 DEFAULT_REGION = "europe-west1"
 GCE_INSTANCE = f"link-artists-{ENV_SHORT_NAME}"
```
Comment (cdelabre) [inline orchestration/dags/jobs/ml/link_artists.py:34]: Possible d'être cohérent entre link-artists / artist_linkage / matched_artists sur les naming des dossiers, du job et MLflow 🙏

File: jobs/ml_jobs/artist_linkage/utils/mlflow.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+def get_mlflow_client_id() -> str:
+    client = secretmanager.SecretManagerServiceClient()
+    name = f"projects/{GCP_PROJECT_ID}/secrets/{MLFLOW_SECRET_NAME}/versions/1"
+    response = client.access_secret_version(name=name)
+    return response.payload.data.decode("UTF-8")
+
+
+def connect_remote_mlflow(
+    client_id: str, mlflow_env: Literal["ehp", "prod"] = "ehp"
```
Comment (cdelabre) [inline jobs/ml_jobs/artist_linkage/utils/mlflow.py:26]: La variable mlflow_env n'est pas utilisée dans la fonction au-dessus. Par ailleurs,  cette variable est connue (ENV_SHORT_NAME). 
On peut avoir une MLFLOW_URI générique via constant et simplifier la logique ici.

File: jobs/ml_jobs/artist_linkage/utils/mlflow.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    GCP_PROJECT_ID,
+    MLFLOW_EHP_URI,
+    MLFLOW_PROD_URI,
+    MLFLOW_SECRET_NAME,
+)
+
+
+def get_mlflow_client_id() -> str:
+    client = secretmanager.SecretManagerServiceClient()
+    name = f"projects/{GCP_PROJECT_ID}/secrets/{MLFLOW_SECRET_NAME}/versions/1"
```
Comment (cdelabre) [inline jobs/ml_jobs/artist_linkage/utils/mlflow.py:20]: 1 -> lastest

File: jobs/ml_jobs/artist_linkage/evaluate.py [hunk 6]
```diff
@@ -1,6 +1,18 @@
...
 import typer
 
+from utils.constants import ENV_SHORT_NAME, EXPERIMENT_BASE_NAME
+from utils.mlflow import (
+    connect_remote_mlflow,
+    get_mlflow_client_id,
+    get_mlflow_experiment,
+)
+
+METRICS_PER_DATASET_CSV_PATH = "metrics_per_dataset.csv"
```
Comment (cdelabre) [inline jobs/ml_jobs/artist_linkage/evaluate.py:13]: PATH -> FILE ?

**Review by cdelabre — COMMENTED** [review]
🫧

File: jobs/ml_jobs/artist_linkage/utils/constants.py (hunk 2 above)
Comment (cdelabre) [reply jobs/ml_jobs/artist_linkage/utils/constants.py:10]: -> Cela permet de lancer le job en adhoc si besoin sans perturber l'experiment schedule. 
+ garder un pseudo versionning si nécéssaire

Comment (lmontier) [reply jobs/ml_jobs/artist_linkage/utils/constants.py:10]: :+1

File: jobs/ml_jobs/artist_linkage/utils/mlflow.py (hunk 4 above)
Comment (lmontier-pass) [reply jobs/ml_jobs/artist_linkage/utils/mlflow.py:26]: :+1:

File: jobs/ml_jobs/artist_linkage/utils/mlflow.py (hunk 5 above)
Comment (lmontier-pass) [reply jobs/ml_jobs/artist_linkage/utils/mlflow.py:20]: :+1

**Review by cdelabre — APPROVED** [review]

**Pull request merged**